psycop/projects/cvd/feature_generation/cohort_definition/eligible_prediction_times/single_filters.py

```diff
@@ -0,0 +1,102 @@
+    load_move_into_rm_for_exclusion,
+)
+from psycop.projects.cvd.feature_generation.cohort_definition.eligible_prediction_times.eligible_config import (
+    AGE_COL_NAME,
+    MIN_AGE,
+    MIN_DATE,
+)
+from psycop.projects.cvd.feature_generation.cohort_definition.outcome_specification.combined import (
+    get_first_cvd_indicator,
+)
+from psycop.projects.t2d.feature_generation.cohort_definition.eligible_prediction_times.add_age import (
+    add_age,
+)
+
+
+class CVDMinDateFilter(PredictionTimeFilter):
+    @staticmethod
+    def apply(df: pl.DataFrame) -> pl.DataFrame:
+        after_df = df.filter(pl.col("timestamp") > MIN_DATE)
+        return after_df
+
+
+class CVDMinAgeFilter(PredictionTimeFilter):
+    @staticmethod
+    def apply(df: pl.DataFrame) -> pl.DataFrame:
+        df = add_age(df)
+        after_df = df.filter(pl.col(AGE_COL_NAME) >= MIN_AGE)
+        return after_df
+
+
+class WithoutPrevalentCVD(PredictionTimeFilter):
+    @staticmethod
+    def apply(df: pl.DataFrame) -> pl.DataFrame:
+        first_cvd_indicator = pl.from_pandas(get_first_cvd_indicator())
+
+        indicator_before_min_date = first_cvd_indicator.filter(
+            pl.col("timestamp") < MIN_DATE,
+        )
+
+        prediction_times_from_patients_with_cvd = df.join(
+            indicator_before_min_date,
+            on="dw_ek_borger",
+            how="inner",
+        )
+
+        no_prevalent_cvd = df.join(
+            prediction_times_from_patients_with_cvd,
+            on="dw_ek_borger",
+            how="anti",
+        )
+
+        return no_prevalent_cvd.drop(["age"])
+
+
+class NoIncidentCVD(PredictionTimeFilter):
+    @staticmethod
+    def apply(df: pl.DataFrame) -> pl.DataFrame:
+        contacts_with_cvd = pl.from_pandas(
+            get_first_cvd_indicator(),
+        )
+
+        contacts_with_hba1c = df.join(
+            contacts_with_cvd,
+            on="dw_ek_borger",
+            how="left",
+            suffix="_result",
+        )
+
+        after_incident_cvd = contacts_with_hba1c.filter(
+            pl.col("timestamp") > pl.col("timestamp_result"),
+        )
+
+        not_after_incident_cvd = contacts_with_hba1c.join(
+            after_incident_cvd,
+            on="dw_ek_borger",
+            how="anti",
+        )
+
+        return not_after_incident_cvd.drop(["timestamp_result"])
+
+
+class CVDWashoutMove(PredictionTimeFilter):
+    @staticmethod
+    def apply(df: pl.DataFrame) -> pl.DataFrame:
+        not_within_two_years_from_move = pl.from_pandas(
+            PredictionTimeFilterer(
+                prediction_times_df=df.to_pandas(),
+                entity_id_col_name="dw_ek_borger",
+                quarantine_timestamps_df=load_move_into_rm_for_exclusion(),
+                quarantine_interval_days=730,
+                timestamp_col_name="timestamp",
+            ).run_filter(),
+        )
+
+        return not_within_two_years_from_move
```

Comment:
@erikperfalk Vil super gerne have din holdning her! Er det læsbart? Nogle fejl du kan se? :-)

(Koden bliver anvendt i `psycop/projects/cvd/feature_generation/cohort_definition/cohort_definition.py`)

Reply:
Det ser fint ud.

Jeg har 2 kommentarer:
- Ift. diagnoserne, så har du ikke taget med alle score2 diagnoser f.eks. hypertensive, arytmier/heart failure? 
- Under class non_incident_cvd, så står det contacts_with_hba1c, men jeg kan ikke se koblingen til det?

Reply:
Tak Erik! 

Ift. SCORE-2, så er det øverste afsnit kun til "fatal cardiovascular disease"; dvs. dødsårsager, hvor en af diagnoserne er markeret som årsagen. Siden vi ikke har dødsårsager, så kan vi ikke bruge det.

Re: `contacts_with_hba1c`, så er det en fejl fra mig, har glemt at omdøbe. Tak!

Reply:
Ja, det kan jeg godt se. Men måske kan man overveje arytmier og hjertesvigt diagnoser til dit outcome?